dart: 3.9.4 -> 3.10.4

r-ryantm opened the pull request:
Automatic update generated by [nixpkgs-update](https://github.com/nix-community/nixpkgs-update) tools. This update was made based on information from passthru.updateScript.

meta.description for dart is: Scalable programming language, with robust libraries and runtimes, for building web, server, and mobile apps

meta.homepage for dart is: https://dart.dev




###### Updates performed
- Ran passthru.UpdateScript

###### To inspect upstream changes





###### Impact

<b>Checks done</b>

---

- built on NixOS
- The tests defined in `passthru.tests`, if any, passed
- found 3.10.0 with grep in /nix/store/83wsi3dz6rkgkf48qlv2y0la61f3nbdm-dart-3.10.0
- found 3.10.0 in filename of file in /nix/store/83wsi3dz6rkgkf48qlv2y0la61f3nbdm-dart-3.10.0

---

<details>
<summary>
<b>Rebuild report</b> (if merged into master) (click to expand)
</summary>

```
42 total rebuild path(s)

42 package rebuild(s)

First fifty rebuilds by attrpath

adw-gtk3
ags
cosmic-settings-daemon
dart
dart-sass
discourse
discourseAllPlugins
domine
expidus.calculator
expidus.file-manager
feishin
flutterPackages-source.stable
flutterPackages-source.v3_29
flutterPackages-source.v3_32
flutterPackages-source.v3_35
flutterPackages-source.v3_38
fvm
gitlab
gitlab-ee
homer
hyprpanel
jekyll
jsonnet
mealie
melos
multipass
pkgsRocm.hyprpanel
protoc-gen-dart
quarto
quartoMinimal
r2modman
redisinsight
rstudio
rstudio-server
rstudioServerWrapper
rstudioWrapper
rustdesk-flutter
shopware-cli
unsure
voicevox
vscode-runner
```

</details>

<details>
<summary>
<b>Instructions to test this update</b> (click to expand)
</summary>

---

Either **download from the cache**:
```
nix-store -r /nix/store/83wsi3dz6rkgkf48qlv2y0la61f3nbdm-dart-3.10.0 \
  --option binary-caches 'https://cache.nixos.org/ https://nixpkgs-update-cache.nix-community.org/' \
  --option trusted-public-keys '
  nixpkgs-update-cache.nix-community.org-1:U8d6wiQecHUPJFSqHN9GSSmNkmdiFW7GW7WNAnHW0SM=
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  '
```
(The nixpkgs-update cache is only trusted for this store-path realization.)
For the cached download to work, your user must be in the `trusted-users` list or you can use `sudo` since root is effectively trusted.

Or, **build yourself**:
```
nix-build -A dart https://github.com/r-ryantm/nixpkgs/archive/b126afb3496c313b2ebd57b1d7d82e24ea2c7ac1.tar.gz
```
Or:
```
nix build github:r-ryantm/nixpkgs/b126afb3496c313b2ebd57b1d7d82e24ea2c7ac1#dart
```

After you've downloaded or built it, look at the files and if there are any, run the binaries:
```
ls -la /nix/store/83wsi3dz6rkgkf48qlv2y0la61f3nbdm-dart-3.10.0
ls -la /nix/store/83wsi3dz6rkgkf48qlv2y0la61f3nbdm-dart-3.10.0/bin
```

---

</details>
<br/>



### Pre-merge build results

We have automatically built all packages that will get rebuilt due to
this change.

This gives evidence on whether the upgrade will break dependent packages.
Note sometimes packages show up as _failed to build_ independent of the
change, simply because they are already broken on the target branch.

:warning: nixpkgs-review took longer than 180m and timed out

---

###### Maintainer pings

cc @GRBurst for [testing](https://github.com/nix-community/nixpkgs-update/blob/main/doc/nixpkgs-maintainer-faq.md#r-ryantm-opened-a-pr-for-my-package-what-do-i-do).

> [!TIP]
> As a maintainer, if your package is located under `pkgs/by-name/*`, you can comment **`@NixOS/nixpkgs-merge-bot merge`** to automatically merge this update using the [`nixpkgs-merge-bot`](https://github.com/NixOS/nixpkgs/blob/master/ci/README.md#nixpkgs-merge-bot).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

r-ryantm commented:
Automatic update generated by [nixpkgs-update](https://github.com/nix-community/nixpkgs-update) tools. This update was made based on information from passthru.updateScript.

meta.description for dart is: Scalable programming language, with robust libraries and runtimes, for building web, server, and mobile apps

meta.homepage for dart is: https://dart.dev




###### Updates performed
- Ran passthru.UpdateScript

###### To inspect upstream changes





###### Impact

<b>Checks done</b>

---

- built on NixOS
- The tests defined in `passthru.tests`, if any, passed
- found 3.10.3 with grep in /nix/store/1c7ajw2i8kqgh1w11s3p29ry4309gs3d-dart-3.10.3
- found 3.10.3 in filename of file in /nix/store/1c7ajw2i8kqgh1w11s3p29ry4309gs3d-dart-3.10.3

---

<details>
<summary>
<b>Rebuild report</b> (if merged into master) (click to expand)
</summary>

```
43 total rebuild path(s)

43 package rebuild(s)

First fifty rebuilds by attrpath

adw-gtk3
ags
bitwarden-desktop
cosmic-settings-daemon
dart
dart-sass
discourse
discourseAllPlugins
domine
expidus.calculator
expidus.file-manager
feishin
flutterPackages-source.stable
flutterPackages-source.v3_29
flutterPackages-source.v3_32
flutterPackages-source.v3_35
flutterPackages-source.v3_38
fvm
gitlab
gitlab-ee
homer
hyprpanel
jekyll
jsonnet
mealie
melos
pkgsRocm.hyprpanel
protoc-gen-dart
quarto
quartoMinimal
r2modman
redisinsight
rstudio
rstudio-server
rstudioServerWrapper
rstudioWrapper
rustdesk-flutter
shopware-cli
tuliprox
unsure
voicevox
vscode-runner
```

</details>

<details>
<summary>
<b>Instructions to test this update</b> (click to expand)
</summary>

---

Either **download from the cache**:
```
nix-store -r /nix/store/1c7ajw2i8kqgh1w11s3p29ry4309gs3d-dart-3.10.3 \
  --option binary-caches 'https://cache.nixos.org/ https://nixpkgs-update-cache.nix-community.org/' \
  --option trusted-public-keys '
  nixpkgs-update-cache.nix-community.org-1:U8d6wiQecHUPJFSqHN9GSSmNkmdiFW7GW7WNAnHW0SM=
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  '
```
(The nixpkgs-update cache is only trusted for this store-path realization.)
For the cached download to work, your user must be in the `trusted-users` list or you can use `sudo` since root is effectively trusted.

Or, **build yourself**:
```
nix-build -A dart https://github.com/r-ryantm/nixpkgs/archive/52e1f2d2594441e36a637d3d9e79cd30426b200a.tar.gz
```
Or:
```
nix build github:r-ryantm/nixpkgs/52e1f2d2594441e36a637d3d9e79cd30426b200a#dart
```

After you've downloaded or built it, look at the files and if there are any, run the binaries:
```
ls -la /nix/store/1c7ajw2i8kqgh1w11s3p29ry4309gs3d-dart-3.10.3
ls -la /nix/store/1c7ajw2i8kqgh1w11s3p29ry4309gs3d-dart-3.10.3/bin
```

---

</details>
<br/>



### Pre-merge build results

We have automatically built all packages that will get rebuilt due to
this change.

This gives evidence on whether the upgrade will break dependent packages.
Note sometimes packages show up as _failed to build_ independent of the
change, simply because they are already broken on the target branch.

:warning: nixpkgs-review took longer than 180m and timed out

---

###### Maintainer pings

cc @GRBurst for [testing](https://github.com/nix-community/nixpkgs-update/blob/main/doc/nixpkgs-maintainer-faq.md#r-ryantm-opened-a-pr-for-my-package-what-do-i-do).

> [!TIP]
> As a maintainer, if your package is located under `pkgs/by-name/*`, you can comment **`@NixOS/nixpkgs-merge-bot merge`** to automatically merge this update using the [`nixpkgs-merge-bot`](https://github.com/NixOS/nixpkgs/blob/master/ci/README.md#nixpkgs-merge-bot).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Hythera reviewed (approved):
`tuliprox` needs #468094 to get merged, to fix that build.

r-ryantm committed:
dart: 3.9.4 -> 3.10.4

qzylinra commented:
Why not merge this? The bitwarden-desktop build works fine.

mdaniels5757 commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 462771`
Commit: `b9ed7e8071c1732dbe3f62479ee734ec25620dc6`

---
### `x86_64-linux`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>bitwarden-desktop</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 46 packages built:</summary>
  <ul>
    <li>adw-gtk3</li>
    <li>ags</li>
    <li>cosmic-settings-daemon</li>
    <li>dart</li>
    <li>dart-sass</li>
    <li>dart-sass.pubcache</li>
    <li>discourse</li>
    <li>discourseAllPlugins</li>
    <li>domine</li>
    <li>domine.pubcache</li>
    <li>feishin</li>
    <li>flutterPackages-source.v3_32</li>
    <li>flutterPackages-source.v3_35</li>
    <li>fvm</li>
    <li>fvm.pubcache</li>
    <li>gitlab</li>
    <li>gitlab-ee</li>
    <li>homer</li>
    <li>hyprpanel</li>
    <li>jekyll</li>
    <li>jsonnet</li>
    <li>jsonnet.doc</li>
    <li>mealie</li>
    <li>mealie.dist</li>
    <li>melos</li>
    <li>melos.pubcache</li>
    <li>protoc-gen-dart</li>
    <li>protoc-gen-dart.pubcache</li>
    <li>quarto</li>
    <li>quartoMinimal</li>
    <li>r2modman</li>
    <li>redisinsight</li>
    <li>rstudio</li>
    <li>rstudio-server</li>
    <li>rstudioServerWrapper</li>
    <li>rstudioWrapper</li>
    <li>rustdesk-flutter</li>
    <li>rustdesk-flutter.debug</li>
    <li>rustdesk-flutter.pubcache</li>
    <li>shopware-cli</li>
    <li>tuliprox</li>
    <li>unsure</li>
    <li>unsure.pubcache</li>
    <li>voicevox</li>
    <li>vscode-runner</li>
    <li>vscode-runner.pubcache</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>bitwarden-desktop</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 46 packages built:</summary>
  <ul>
    <li>adw-gtk3</li>
    <li>ags</li>
    <li>cosmic-settings-daemon</li>
    <li>dart</li>
    <li>dart-sass</li>
    <li>dart-sass.pubcache</li>
    <li>discourse</li>
    <li>discourseAllPlugins</li>
    <li>domine</li>
    <li>domine.pubcache</li>
    <li>feishin</li>
    <li>flutterPackages-source.v3_32</li>
    <li>flutterPackages-source.v3_35</li>
    <li>fvm</li>
    <li>fvm.pubcache</li>
    <li>gitlab</li>
    <li>gitlab-ee</li>
    <li>homer</li>
    <li>hyprpanel</li>
    <li>jekyll</li>
    <li>jsonnet</li>
    <li>jsonnet.doc</li>
    <li>mealie</li>
    <li>mealie.dist</li>
    <li>melos</li>
    <li>melos.pubcache</li>
    <li>protoc-gen-dart</li>
    <li>protoc-gen-dart.pubcache</li>
    <li>quarto</li>
    <li>quartoMinimal</li>
    <li>r2modman</li>
    <li>redisinsight</li>
    <li>rstudio</li>
    <li>rstudio-server</li>
    <li>rstudioServerWrapper</li>
    <li>rstudioWrapper</li>
    <li>rustdesk-flutter</li>
    <li>rustdesk-flutter.debug</li>
    <li>rustdesk-flutter.pubcache</li>
    <li>shopware-cli</li>
    <li>tuliprox</li>
    <li>unsure</li>
    <li>unsure.pubcache</li>
    <li>voicevox</li>
    <li>vscode-runner</li>
    <li>vscode-runner.pubcache</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>rstudio</li>
    <li>rstudioWrapper</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 28 packages built:</summary>
  <ul>
    <li>adw-gtk3</li>
    <li>bitwarden-desktop</li>
    <li>dart</li>
    <li>dart-sass</li>
    <li>dart-sass.pubcache</li>
    <li>domine</li>
    <li>domine.pubcache</li>
    <li>feishin</li>
    <li>fvm</li>
    <li>fvm.pubcache</li>
    <li>homer</li>
    <li>jekyll</li>
    <li>jsonnet</li>
    <li>jsonnet.doc</li>
    <li>mealie</li>
    <li>mealie.dist</li>
    <li>melos</li>
    <li>melos.pubcache</li>
    <li>protoc-gen-dart</li>
    <li>protoc-gen-dart.pubcache</li>
    <li>quarto</li>
    <li>quartoMinimal</li>
    <li>r2modman</li>
    <li>shopware-cli</li>
    <li>tuliprox</li>
    <li>unsure</li>
    <li>unsure.pubcache</li>
    <li>voicevox</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>rstudio</li>
    <li>rstudioWrapper</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 28 packages built:</summary>
  <ul>
    <li>adw-gtk3</li>
    <li>bitwarden-desktop</li>
    <li>dart</li>
    <li>dart-sass</li>
    <li>dart-sass.pubcache</li>
    <li>domine</li>
    <li>domine.pubcache</li>
    <li>feishin</li>
    <li>fvm</li>
    <li>fvm.pubcache</li>
    <li>homer</li>
    <li>jekyll</li>
    <li>jsonnet</li>
    <li>jsonnet.doc</li>
    <li>mealie</li>
    <li>mealie.dist</li>
    <li>melos</li>
    <li>melos.pubcache</li>
    <li>protoc-gen-dart</li>
    <li>protoc-gen-dart.pubcache</li>
    <li>quarto</li>
    <li>quartoMinimal</li>
    <li>r2modman</li>
    <li>shopware-cli</li>
    <li>tuliprox</li>
    <li>unsure</li>
    <li>unsure.pubcache</li>
    <li>voicevox</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>bitwarden-desktop</summary>
<pre>[Main]       ./src/key-management/biometrics/os-biometrics-linux.service.ts 9.72 KiB [built] [code generated]
[Main]   + 20 modules
[Main]   main (webpack 5.103.0) compiled successfully in 19501 ms
[Main] npm run build:main exited with code 0
npm error Lifecycle script `build` failed with error:
npm error code 1
npm error path /build/source/apps/desktop
npm error workspace @bitwarden/desktop@2025.12.0
npm error location /build/source/apps/desktop
npm error command failed
npm error command sh -c concurrently -n Main,Rend,Prel -c yellow,cyan &quot;npm run build:main&quot; &quot;npm run build:renderer&quot; &quot;npm run build:preload&quot;

ERROR: `npm build` failed

Here are a few things you can try, depending on the error:
1. Make sure your build script (build) exists
  If there is none, set `dontNpmBuild = true`.
2. If the error being thrown is something similar to &quot;error:0308010C:digital envelope routines::unsupported&quot;, add `NODE_OPTIONS = &quot;--openssl-legacy-provider&quot;` to your derivation
  See https://github.com/webpack/webpack/issues/14532 for more information.
</pre>
</details>
</details>

---
<details>
<summary>Error logs: `aarch64-linux`</summary>
<details>
<summary>bitwarden-desktop</summary>
<pre>[Main]       ./src/key-management/biometrics/os-biometrics-linux.service.ts 9.72 KiB [built] [code generated]
[Main]   + 20 modules
[Main]   main (webpack 5.103.0) compiled successfully in 37634 ms
[Main] npm run build:main exited with code 0
npm error Lifecycle script `build` failed with error:
npm error code 1
npm error path /build/source/apps/desktop
npm error workspace @bitwarden/desktop@2025.12.0
npm error location /build/source/apps/desktop
npm error command failed
npm error command sh -c concurrently -n Main,Rend,Prel -c yellow,cyan &quot;npm run build:main&quot; &quot;npm run build:renderer&quot; &quot;npm run build:preload&quot;

ERROR: `npm build` failed

Here are a few things you can try, depending on the error:
1. Make sure your build script (build) exists
  If there is none, set `dontNpmBuild = true`.
2. If the error being thrown is something similar to &quot;error:0308010C:digital envelope routines::unsupported&quot;, add `NODE_OPTIONS = &quot;--openssl-legacy-provider&quot;` to your derivation
  See https://redirect.github.com/webpack/webpack/issues/14532 for more information.
</pre>
</details>
</details>

mdaniels5757 commented:
Full build log for bitwarden-desktop: [bitwarden-desktop-x86_64-linux.log](https://github.com/user-attachments/files/24201151/bitwarden-desktop-x86_64-linux.log)

SuperSandro2000 commented:
Its not nice, that dart has so many unrelated files under $out/bin

```
 ➜ ls result/bin/
 dart   dartaotruntime   dartvm   dartvm.sym   resources   snapshots   utils
```

SuperSandro2000 commented:
I think it might fail to build because of https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/bi/bitwarden-desktop/package.nix#L134

mdaniels5757 commented:
> How many cores does your builder has?

8 for my laptop, the community builder has 24.

mdaniels5757 commented:
> How many cores does your builder has?

@SuperSandro2000 You were on the right track here. #471830

SuperSandro2000 committed:
bitwarden-desktop: use the vendored dart from sass-embedded

Other dart versions might not work, like when we upgraded dart from
3.9.3 to 3.10.4 and the build might even silently fail.

amarshall commented:
If we’re using prebuilt binary now then should set sourceProvenance.

SuperSandro2000 commented:
> If we’re using prebuilt binary now then should set sourceProvenance.

We are using the same prebuilt sass binary as before, we are just not executing it with our dart anymore but with the dart runtime shipped with it.
